app/Helpers/database/leaderboard.php

```diff
@@ -743,22 +775,27 @@ function duplicateLeaderboard(int $gameID, int $leaderboardID, int $duplicateNum
 
 function requestResetLB(int $lbID): bool
 {
-    if ($lbID == 0) {
-        return false;
-    }
+    // TODO remove when LeaderboardEntry is dropped
+    $legacyEntries = LeaderboardEntryLegacy::where('LeaderboardID', $lbID);
+    $legacyDeleted = $legacyEntries->delete();
+    // TODO end remove when LeaderboardEntry is dropped
 
-    $query = "DELETE FROM LeaderboardEntry
-              WHERE LeaderboardID = $lbID";
-    $dbResult = s_mysql_query($query);
+    $entries = LeaderboardEntry::where('leaderboard_id', $lbID);
+    $entriesDeleted = $entries->delete();
 
-    return $dbResult !== false;
+    // When `delete()` returns false, it indicates an error has occurred.
+    return $entriesDeleted !== false && $legacyDeleted !== false;
 }
 
 function requestDeleteLB(int $lbID): bool
 {
-    $query = "DELETE FROM LeaderboardDef WHERE ID = $lbID";
+    $leaderboard = Leaderboard::find($lbID);
 
-    $dbResult = s_mysql_query($query);
+    if (!$leaderboard) {
+        return false;
+    }
+
+    $leaderboard->delete();
```

Comment:
This might need to be a `forceDelete` too. After deleting a leaderboard, it still appears in the list, but cannot be deleted again.

Reply:
Yep, given the nature of the previous code that was there, you're right. Good catch, thanks.